### ISSUE TITLE
WorldServer.saveChunkData broken in 1.8 and 1.9

### ISSUE THREAD

**Issue by alexbobp** [MinecraftForge/MinecraftForge#2985]
Tested with: forge-1.8.9-11.15.1.1902-1.8.9-universal.jar

WorldServer.saveChunkData caused infinite looping messages like:
`[23:31:44] [Server thread/INFO]: ThreadedAnvilChunkStorage (world): All chunks are saved`
The server console becomes unresponsive.  The stop command does not work and ctrl-c is necessary.

To reproduce: stand up a forge server (no mods necessary) and issue the command:
`save-all flush`

Expected result (from vanilla 1.8.9 server):
```
> save-all flush
[23:34:40] [Server thread/INFO]: Saving...
[23:34:40] [Server thread/INFO]: commands.save.flushStart
[23:34:40] [Server thread/INFO]: ThreadedAnvilChunkStorage (world): All chunks are saved
[23:34:40] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[23:34:40] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[23:34:40] [Server thread/INFO]: commands.save.flushEnd
[23:34:40] [Server thread/INFO]: Saved the world

```

**Comment by covers1624** [comment]
Can confirm forge 1.9.4-12.17.0.1960 is effected.

**Comment by LexManos** [comment]
Humm this is a fernflower decompile issue.
I'll see what I can do to write up a test case and fix it eventually but I can build a quick patch for it.

**Comment by alexbobp** [comment]
Thanks Lex!  Will this fix go to both 1.8 and 1.9?  If you can update this bug later with fix versions I will appreciate that so I can make my mod decide whether it needs to do the workaround.

**Comment by LatvianModder** [comment]
So That's why my backups didn't work back in 1.7.10... And I thought I'm doing something wrong ¯\_(ツ)_/¯

**Comment by alexbobp** [comment]
I don't think so Latvian.  I did not find this problem at all in 1.7.10

**Comment by LexManos** [comment]
It'd effect all versions of MC with the /save-all flush command.

**Comment by alexbobp** [comment]
Not in my testing Lex.  save-all flush works on my forge 1.7.10 servers as well as when I tested with gradlew runServer

**Comment by LexManos** [comment]
Either way it's a fernflower decompile issue and has been patched by forge. It's functional now.

**Comment by alexbobp** [comment]
Will there be a fix for 1.8.9?  It seems only 1.9.4 got the fix.

**Comment by TehNut** [comment]
1.8.9 is EOL in terms of Forge. 1.9.4 is the only actively developed version.

**Comment by alexbobp** [comment]
Yeah but we're not talking about a feature enhancement request.  This is a basic server management feature in vanilla, that forge broke.  Are you telling me the strategy for encouraging people to upgrade requires that we withhold simple fixes to severe bugs?

**Comment by NikkyAI** [comment]
still happens in 1.8.9-11.15.1.2318
which was released about a year after the bug ewas reported.. so much about EOL
source: timestamps on https://files.minecraftforge.net/maven/net/minecraftforge/forge/index_1.8.9.html

**Comment by TehNut** [comment]
That doesn't mean this particular fix was pulled into the 1.8.9 branch. The only reason it was built was because there was a *major* exploit.

**Comment by LexManos** [comment]
If @mezz feels like making a build sure. But yes 1.8.9 is EOL. Simple as that. People have been dealing with this for years they can keep dealing aith the quirks of using legacy software if they are choosing to use old code.
